input/script_utils.go

```diff
@@ -1822,29 +1824,46 @@ func NewLocalCommitScriptTree(csvTimeout uint32,
 		return nil, err
 	}
 
-	// With the delay script computed, we'll now create a tapscript tree
-	// with a single leaf, and then obtain a root from that.
-	tapLeaf := txscript.NewBaseTapLeaf(delayScript)
-	tapScriptTree := txscript.AssembleTaprootScriptTree(tapLeaf)
+	// Next, we'll need to construct the revocation path, which is just a
+	// simple checksig script.
+	builder = txscript.NewScriptBuilder()
+	builder.AddData(schnorr.SerializePubKey(revokeKey))
+	builder.AddOp(txscript.OP_CHECKSIG)
+
+	revokeScript, err := builder.Script()
+	if err != nil {
+		return nil, err
+	}
+
+	// With both scripts computed, we'll now create a tapscript tree with
+	// the two leaves, and then obtain a root from that.
+	delayTapLeaf := txscript.NewBaseTapLeaf(delayScript)
+	revokeTapLeaf := txscript.NewBaseTapLeaf(revokeScript)
+	tapScriptTree := txscript.AssembleTaprootScriptTree(
+		delayTapLeaf, revokeTapLeaf,
+	)
 	tapScriptRoot := tapScriptTree.RootNode.TapHash()
 
 	// Now that we have our root, we can arrive at the final output script
 	// by tweaking the internal key with this root.
 	toLocalOutputKey := txscript.ComputeTaprootOutputKey(
-		revokeKey, tapScriptRoot[:],
+		&TaprootNUMsKey, tapScriptRoot[:],
```

Comment:
Doesn't this need to be the `<local_delayedpubkey>` instead so it is revealed in the control block (for the "anyone can sweep the anchor after 16 blocks" use case)?

Another thought I had while reviewing this: We could also use the annex to push the `<local_delayedpubkey>`. Though I think the BIP mentions the annex shouldn't be used for witness v1, though I'm not sure if that's enforced in Bitcoin Core. Also, since the annex is witness only, it cannot be enforced and implementations might choose not to do it to save some fees...

Reply:
See the spec comments, we reveal the delay key in a data push now.